File: result/translate/transcoder/python2java/java/FORM_MINIMUM_NUMBER_FROM_GIVEN_SEQUENCE_1.java
public static String f_gold ( String seq ) {
  int n = seq . length ( ) ;
  if ( ( n >= 9 ) && ( seq . charAt ( n ) == 'I' ) ) {
    return "-1" ;
  }
  StringBuilder sb = new StringBuilder ( n + 1 ) ;
  int count = 1 ;
  for ( int i = 0 ;
  i < n + 1 ;
  i ++ ) {
    if ( ( i == n || seq . charAt ( i ) == 'I' ) && ( seq . charAt ( i + 1 ) == 'I' ) ) {
      for ( int j = i - 1 ;
      j >= 0 ;
      j -- ) {
        sb . append ( ( char ) ( '0' + count ) ) ;
        count ++ ;
        if ( ( j >= 0 && seq . charAt ( j ) == 'I' ) && ( seq . charAt ( j + 1 ) == 'I' ) ) {
          break ;
        }
      }
    }
  }
  return sb . toString ( ) ;
}

File: result/translate/transcoder/java2python/python/FORM_MINIMUM_NUMBER_FROM_GIVEN_SEQUENCE_1.py
def f_gold ( seq ) :
    n = len ( seq )
    if n >= 9 :
        return "-1"
    result = [ ]
    count = 1
    for i in range ( 0 , n ) :
        if i == n or seq [ i ] == 'I' :
            for j in range ( i - 1 , - 1 , - 1 ) :
                result.append ( chr ( int ( '0' + count ) ) )
                if j >= 0 and seq [ j ] == 'I' :
                    break
    return ''.join ( result )
